labman/db/tests/tester.py

```diff
@@ -224,6 +229,175 @@ def integration_tests():
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # gDNA extraction
+    print('\tgDNA extraction process', end='', flush=True)
+    ext_processes = []
+    for _, sp in sp_processes:
+        ext_processes.append(create_gdna_extraction_process(user, sp))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # gDNA compression
+    print('\tgDNA compression process', end='', flush=True)
+    comp_processes = []
+    for pos in range(0, num_plates, 4):
+        plates = [gp for _, gp in ext_processes[pos:pos + 4]]
+        comp_processes.append(create_compression_process(user, plates))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # gDNA compressed quantification
+    print('\tcompressed gDNA quantification process', end='', flush=True)
+    gdna_comp_quant_processes = []
+    for _, cp in comp_processes:
+        gdna_comp_quant_processes.append(
+            create_quantification_process(user, cp))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Normalization process
+    print('\tgDNA normalization process', end='', flush=True)
+    norm_processes = []
+    for qp in gdna_comp_quant_processes:
+        norm_processes.append(create_normalization_process(user, qp))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Library prep shotgun
+    print('\tShotgun library prep process', end='', flush=True)
+    shotgun_processes = []
+    for _, norm_plate in norm_processes:
+        shotgun_processes.append(create_shotgun_process(user, norm_plate))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Quantify library plate
+    print('\tShotgun library quantification process', end='', flush=True)
+    shotgun_quant_processes = []
+    for _, sp in shotgun_processes:
+        shotgun_quant_processes.append(create_quantification_process(user, sp))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Pooling process
+    print('\tPooling process', end='', flush=True)
+    pool_processes = []
+    for sqp, (_, sp) in zip(shotgun_quant_processes, shotgun_processes):
+        pool_processes.append(
+            create_plate_pool_process(
+                user, sqp, sp, {'function': 'equal',
+                                'parameters': {'total_vol': 60, 'size': 500}}))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Sequencing process
+    print('\tSequencing process', end='', flush=True)
+    for pool_process in pool_processes:
+        create_sequencing_process(user, [pool_process.pool])
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+
+@tester.command()
+@click.option('--num_plates', required=False, type=click.IntRange(1, None),
+              default=10, show_default=True,
+              help='Number of plates to create per workflow')
+def stress_tests(num_plates):
+    samples = get_samples()
```

Comment:
So just as a refresher, this is going to grab each of the samples that currently exist in the Qiita db, and add them sequentially to plates -- maintaining study associations? Since there's only one study currently in my test db it looks like it's just cycling through those samples.

Reply:
This is getting the list of samples in the entire DB. The stress_test command loops over all the samples in the DB. The sample id is linked to a study in a table described in the Qiita repository `qiita.study_sample`, and the table `qiita.sample_composition` has a FK to that table (and hence maintaining the relation study-sample). Since we want to be able to add, let's say, 100 plates in the DB (I did this on my DB to stress test my previous PR) the `stress_tests` function loops over all the samples in the DB. If we run this in the maptest system, it will be looping through all the samples in the DB.